res_company_code/models/ir_http.py

```diff
@@ -13,24 +13,22 @@ def session_info(self):
         result = super().session_info()
         user = request.env.user
 
-        display_switch_company_menu = (
-            user.has_group("base.group_multi_company") and len(user.company_ids) > 1
-        )
+        if result.get("display_switch_company_menu"):
+            # Replace company name by company complete name in the session
+            # The values are used in the switch_company_menu widget (web module)
+            result["user_companies"]["current_company"] = user.company_id.id
```

Comment:
This is the line that errors. Reproducible running stock odoo unit tests with `--init mail,res_company_code`. The `user_companies` value is not set for users without group `base.group_user`.

Reply:
Covered with unit tests.